src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -1065,11 +1081,31 @@ private UsernamePasswordAuthenticationToken loginAndSetUserData(
             user.addProperty(new Mailer.UserProperty(email));
         }
 
+        LOGGER.fine("Fullname expr " + fullNameFieldExpr);
         String fullName = determineStringField(fullNameFieldExpr, idToken, userInfo);
         if (fullName != null) {
             user.setFullName(fullName);
         }
 
+        // Determine avatar if set
+        if (StringUtils.isNotBlank(avatarFieldName)) {
+            try {
+                OicAvatarProperty.AvatarImage avatarImage = new OicAvatarProperty.AvatarImage("image/png");
```

Comment:
only png supported?

Reply:
I would need to probe the content type from the byte I guess. I just get the binary as is from my LDAP, then keycloak store it as base64 encoded (then added to the profile endpoint). Not sure if the content type is stored anywhere, I will investigate this.